tests/TransactionTest.php

```diff
@@ -0,0 +1,226 @@
+        $check2 = User::on($this->connection)->find($user2->_id);
+        $this->assertEquals($new_age, $check1->age);
+        $this->assertEquals($new_age, $check2->age);
+    }
+
+    public function testDelete()
+    {
+        /** rollback test */
+        $user1 = User::on($this->connection)->create($this->insertData);
+        DB::beginTransaction();
+        $user1->delete();
+        DB::rollBack();
+        $check1 = User::on($this->connection)->find($user1->_id);
+        $this->assertNotNull($check1);
+
+        /** commit test */
+        User::on($this->connection)->truncate();
+        $user1 = User::on($this->connection)->create($this->insertData);
+        DB::beginTransaction();
+        $user1->delete();
+        DB::commit();
+        $check1 = User::on($this->connection)->find($user1->_id);
+        $this->assertNull($check1);
+    }
+
+    public function testTransaction(): void
+    {
+        $count = User::on($this->connection)->count();
+        $this->assertEquals(1, $count);
+        $newAge = $this->originData['age'] + 1;
+
+        DB::transaction(function () use ($newAge) {
+            User::on($this->connection)->create($this->insertData);
+            User::on($this->connection)->where($this->originData['name'])->update(['age' => $newAge]);
+        });
+
+        $count = User::on($this->connection)->count();
+        $this->assertEquals(2, $count);
+
+        $checkInsert = User::on($this->connection)->where($this->insertData['name'])->first();
+        $this->assertNotNull($checkInsert);
+
+        $checkUpdate = User::on($this->connection)->where($this->originData['name'])->first();
+        $this->assertEquals($newAge, $checkUpdate->age);
+    }
+
+    public function testIsTransactionSuccessfullyAbortedWhenAttemptsEnd(): void
+    {
+        $oldUsersCount = User::on($this->connection)->count();
+
+        $result = DB::transaction(function () {
+            User::on($this->connection)->create($this->insertData);
+            User::on($this->connection)->create($this->insertData);
+        }, 0);
+
+        $newUsersCount = User::on($this->connection)->count();
+
+        $this->assertNull($result);
+        $this->assertEquals($newUsersCount, $oldUsersCount);
+    }
+
+    public function testIsDataInTransactionReturnedWhenTransactionWasSuccess(): void
+    {
+        $oldUsersCount = User::on($this->connection)->count();
+
+        $result = DB::transaction(function () {
+            return User::on($this->connection)->create($this->insertData);
+        });
+
+        $newUsersCount = User::on($this->connection)->count();
+
+        $this->assertNotNull($result);
+        $this->assertEquals($result->title, $this->insertData['title']);
+        $this->assertNotEquals($newUsersCount, $oldUsersCount);
+    }
+
+    public function testThrowExceptionForNestedTransactions(): void
+    {
+        $this->expectException(RuntimeException::class);
+        $this->expectExceptionCode(53 /* InvalidIdField */);
```

Comment:
Contrast this with my earlier comment in `TransactionBuilderTest::TestNestingTransaction()`. Why does this method expect an exception but not the other?

Reply:
Already fixed